tests/integration/test_locking.py

```diff
@@ -0,0 +1,80 @@
+import logging
+import time
+import threading
+
+import pytest
+
+from tests.bento_service_examples.example_bento_service import ExampleBentoService
+import subprocess
+
+logger = logging.getLogger('bentoml.test')
+
+
+class ThreadWithResult(threading.Thread):
+    def __init__(
+        self, group=None, target=None, name=None, args=(), kwargs={}, *, daemon=None
+    ):
+        def function():
+            self.result = target(*args, **kwargs)
+
+        super().__init__(group=group, target=function, name=name, daemon=daemon)
+
+
+def run_delayed_thread(t1, t2):
+    t1.start()
+    time.sleep(1)
```

Comment:
anything less than 0.5s here causes some instability in the guarantee that t1 acquires lock before t2 here (related to concern above)